functions/__sf_section_dir.fish

```diff
@@ -23,7 +23,8 @@ function __sf_section_dir -d "Display the current truncated directory"
 	set -l dir
 
 	if test $SPACEFISH_DIR_TRUNC_REPO = true -a (__sf_util_git_branch)
-		set -l git_root (git rev-parse --show-toplevel)
+		# Derive repo root from its git directory
+		set -l git_root (string replace '/.git' '' (git rev-parse --absolute-git-dir))
```

Comment:
This requires at least `git v2.13.0`. Because `--absolute-git-dir` [was added with that release.](https://github.com/git/git/commit/a2f5a8762693d5af6dbbca714d882a21d7ba0b75)